spring-web/src/main/java/org/springframework/http/converter/ResourceHttpMessageConverter.java

```diff
@@ -152,22 +152,16 @@ protected void writeContent(Resource resource, HttpOutputMessage outputMessage)
 		// We cannot use try-with-resources here for the InputStream, since we have
 		// custom handling of the close() method in a finally-block.
 		try {
-			InputStream in = resource.getInputStream();
-			try {
-				OutputStream out = outputMessage.getBody();
-				in.transferTo(out);
-				out.flush();
-			}
-			catch (NullPointerException ex) {
-				// ignore, see SPR-13620
-			}
-			finally {
+			try (InputStream in = resource.getInputStream()) {
 				try {
-					in.close();
-				}
-				catch (Throwable ex) {
-					// ignore, see SPR-12999
```

Comment:
Please revert. This will catch all Throwable thrown by the code block, not just when the resource is closed.